fw/tls.c

```diff
@@ -262,7 +263,18 @@ tfw_tls_encrypt(struct sock *sk, struct sk_buff *skb, unsigned int mss_now,
 	struct scatterlist sg[AUTO_SEGS_N], out_sg[AUTO_SEGS_N];
 	struct page **pages = NULL, **pages_end, **p;
 	struct page *auto_pages[AUTO_SEGS_N];
+	bool skb_was_added = false;
 
+	/*
+	 * We should add sk overhead before adding new data to skb which
+	 * already inserted in socket write queue. Because we we fail to
+	 * add overhead after inserting new data to such skb we catch
+	 * warings or bugs in kernel code later.
```

Comment:
I guess this happens because we modify skb and don't return its initial size on the function failure. In this context playing with the overhead looks like a dirty workaround: we don't have any real data in the connection, but large overhead.